blockchain/test_blockchain.go

```diff
@@ -0,0 +1,273 @@
+
+func NewClass(t *testing.T, path string) TestClass {
+	t.Helper()
+
+	snClass, compiledClass, class := classFromFile(t, path)
+	classHash, err := class.Hash()
+	require.NoError(t, err)
+
+	return TestClass{
+		t:        t,
+		class:    class,
+		hash:     *classHash,
+		snClass:  snClass,
+		compiled: compiledClass,
+	}
+}
+
+func (b *TestClass) Hash() felt.Felt {
+	return b.hash
+}
+
+func (b *TestClass) SNClass() *starknet.SierraDefinition {
+	return b.snClass
+}
+
+func (b *TestClass) CompiledClass() *starknet.CompiledClass {
+	return b.compiled
+}
+
+func (b *TestClass) AddAccount(address, balance felt.Felt) {
+	b.accounts = append(b.accounts, &testAccount{
+		t:       b.t,
+		address: address,
+		balance: balance,
+	})
+}
+
+func (b *testBlockchain) Prepare(t *testing.T, classes []TestClass) {
+	t.Helper()
+
+	newClasses := make(map[felt.Felt]core.Class, len(classes))
+	deployedContracts := make(map[felt.Felt]*felt.Felt) // use pointer because of DeployedContracts field
+	balances := make(map[felt.Felt]*felt.Felt)          // use pointer because of StorageDiff field
+
+	for _, class := range classes {
+		newClasses[class.hash] = class.class
+		for _, account := range class.accounts {
+			deployedContracts[account.address] = &class.hash
+			b.addr2classHash[account.address] = class.hash
+			if balance := account.balance; (balance != felt.Felt{}) {
+				balances[account.BalanceKey()] = &balance
+			}
+		}
+	}
+
+	height, err := b.Height()
+	require.NoError(t, err)
+	parentBlock, err := b.BlockByNumber(height)
+	require.NoError(t, err)
+
+	stateUpdate := &core.StateUpdate{
+		OldRoot: parentBlock.GlobalStateRoot,
+		StateDiff: &core.StateDiff{
+			DeployedContracts: deployedContracts,
+			StorageDiffs: map[felt.Felt]map[felt.Felt]*felt.Felt{
+				b.erc20.accounts[0].address: balances,
+				b.erc20.accounts[1].address: balances,
+			},
+		},
+	}
+
+	stateUpdate.NewRoot = b.NewRoot(t, parentBlock.Number+1, stateUpdate, newClasses)
+
+	block := &core.Block{
+		Header: &core.Header{
+			Hash:             utils.HexToFelt(t, "0xb01"),
+			ParentHash:       parentBlock.Hash,
+			Number:           parentBlock.Number + 1,
+			GlobalStateRoot:  stateUpdate.NewRoot,
+			SequencerAddress: utils.HexToFelt(t, "0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8"),
+			Timestamp:        parentBlock.Timestamp + 1,
+			ProtocolVersion:  parentBlock.ProtocolVersion,
+			L1GasPriceETH:    utils.HexToFelt(t, "0x1"),
+			L1GasPriceSTRK:   utils.HexToFelt(t, "0x2"),
+			L1DAMode:         core.Blob,
+			L1DataGasPrice:   &core.GasPrice{PriceInWei: utils.HexToFelt(t, "0x2"), PriceInFri: utils.HexToFelt(t, "0x2")},
+			L2GasPrice:       &core.GasPrice{PriceInWei: utils.HexToFelt(t, "0x1"), PriceInFri: utils.HexToFelt(t, "0x1")},
+		},
+	}
+
+	require.NoError(t, b.Store(block, &core.BlockCommitments{}, stateUpdate, newClasses))
+}
+
+func classFromFile(t *testing.T, path string) (*starknet.SierraDefinition, *starknet.CompiledClass, *core.Cairo1Class) {
```

Comment:
change the order of results to `Cairo1Class`, `SierraDefinition` and `CompiledClass`, that's the compilation order: source file -> sierra file -> casm file.